### PR TITLE
Neaten assessment form side navigation

### PR DESCRIPTION
**Pull request by martha**
## Description

This should fix the appearance of the awkward empty box as shown in screenshots here: https://www.pivotaltracker.com/n/projects/2591838/stories/186976233/comments/240342546

Let me know if the refactor of the logic from `DeleteAssessmentButton.tsx` to here looks okay to you.

## Type of change
[//]: # 'remove options that are not relevant'
- [x] Bug fix
- [ ] New feature (adds functionality)
- [ ] Code clean-up / housekeeping
- [ ] Dependency update

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (eslint)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: src/modules/assessments/components/AssessmentFormSideBar.tsx [hunk 1]
```diff
@@ -60,6 +61,32 @@ const AssessmentFormSideBar: React.FC<Props> = ({
     [enrollment, navigate]
   );
 
+  const shouldShowDeleteButton = (assessment: FullAssessmentFragment) => {
```
Comment (gigxz) [inline src/modules/assessments/components/AssessmentFormSideBar.tsx]: This feels funky to me. It's a function that is defined inside the component that accepts a variable `assessment` that "shadows" the outer variable `assessment` that is a prop. But we always pass the `assessment`(prop) in as the `assessment` var to the function.

I would suggest a different pattern.. Here are some options, I think I prefer the first:

1) Contain this logic inside a useMemo hook. It only needs to re-run when the assessment changes (as opposed to on every render)
```
const showDeleteAssessmentButton = useMemo(() => {
    if (!assessment) return false;

   // full logic for whether deletion is allowed goes here
   return true
}, [assessment]);
```

2) Move this `shouldShowDeleteButton ` function outside of the react component. There is no need for it to be re-defined and called every render, if it just a function evaluating some conditions about an assessment record.

**Review by gigxz — COMMENTED** [review]
Look good to me! small suggestion on react style

File: src/modules/assessments/components/AssessmentFormSideBar.tsx (hunk 1 above)
Comment (martha) [reply src/modules/assessments/components/AssessmentFormSideBar.tsx]: Thanks -- this is much better! :)

File: src/modules/assessments/components/AssessmentFormSideBar.tsx [hunk 2]
```diff
@@ -79,35 +105,45 @@ const AssessmentFormSideBar: React.FC<Props> = ({
...
+                    clientId: assessment.enrollment.client.id,
+                    enrollmentId: assessment.enrollment.id,
+                    assessmentId: assessment.id,
+                  }
+                )}
+              >
+                Print
+              </PrintViewButton>
+            )}
+            {showDeleteAssessmentButton && assessment && (
```
Comment (martha) [inline src/modules/assessments/components/AssessmentFormSideBar.tsx:131]: I added `&& assessment` here to keep typescript happy, even though it's always true from `showDeleteAssessmentButton`. The other thing we could do is pass `assessment as FullAssessmentFragment`. Let me know if you prefer one over the other!

Comment (gigxz) [reply src/modules/assessments/components/AssessmentFormSideBar.tsx:131]: I usually take the approach you have here to keep typescript happy when necessary

**Review by gigxz — APPROVED** [review]

**Pull request merged**